nabs/plan_stubs.py

```diff
@@ -67,3 +68,104 @@ def measure_average(detectors, num, delay=None, stream=None):
     yield from stub_wrapper(count(detectors, num=num, delay=delay))
     # Return the measured average as a dictionary for use in adaptive plans
     return stream.last_event
+
+
+def update_sample(sample_name, path, n_shots):
+    """
+    Update the current sample information after a run.
+
+    Updates the `status` values of each target in the sample,
+    from `False` to `True` to indicate that it is shot.
+
+    Parameters
+    ----------
+    sample_name : str
+        A name to identify the sample grid, should be snake_case style.
+    path : str
+        Path to the `.yml` file. Defaults to the path defined when
+        creating this object.
+    n_shots : int
+        Indicates how many targets have been shot.
+
+    """
+    info = get_sample_targets(sample_name, path)
+    data = {}
+    # list of dictionaries
+    xx = info[0]
+    yy = info[1]
+    # find the index of the targets that is next to be shot
+    x_index = next((index for (index, d) in enumerate(xx)
+                    if d["status"] is False), None)
+    if x_index is None:
+        raise IndexError('Could not get a target index that has not been shot,'
+                         ' probably all targets were shot from this sample?')
+
+    temp_x, temp_y = [], []
+    for i in range(n_shots):
```

Comment:
I am trying to update the saved sample information here, and the way I do it is by knowing how many targets have been shot, and rewrite the information for those targets with the new Status - this can be tricky because I need to make sure I don't add additional targets to the file or I don't take any out or I don't override the wrong ones. Testing this seems like it works fine - but at first glance, are there any edge cases or concerns that I did not consider yet?

Reply:
I can't really follow this part of the code- but I don't see any red flags